ccdaservice/ccda_gateway.php

```diff
@@ -0,0 +1,117 @@
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @package OpenEMR
+ * @author Jerry Padgett <sjpadgett@gmail.com>
+ * @link http://www.open-emr.org
+ */
+//authencate for portal or main- never know where it gets used
+session_start();
+if ( isset($_SESSION['pid']) && isset($_SESSION['patient_portal_onsite_two']) ) {
+	$pid = $_SESSION['pid'];
+	$ignoreAuth = true;
+	$sanitize_all_escapes = true;
+	$fake_register_globals = false;
+	require_once ( dirname( __FILE__ ) . "/../interface/globals.php" );
+	define('IS_DASHBOARD', false);
+	define('IS_PORTAL', $_SESSION['pid']);
+}
+else {
+	session_destroy();
+	$ignoreAuth = false;
+	$sanitize_all_escapes = true;
+	$fake_register_globals = false;
+	require_once ( dirname( __FILE__ ) . "/../interface/globals.php" );
+	if ( ! isset($_SESSION['authUserID']) ){
+		$landingpage = "index.php";
+		header('Location: '.$landingpage);
+		exit;
+	}
+	define('IS_DASHBOARD', $_SESSION['authUserID']);
+	define('IS_PORTAL', false);
+}
+// give me something to do.
+$dowhat = isset($_REQUEST['action']) ? $_REQUEST['action'] : '';
+if( $dowhat && $GLOBALS['ccda_alt_service_enable'] >= 1 ){ // do I need this?
+	require_once ("./../ccdaservice/ssmanager.php");
+	if(!runCheck()){ // woops, try again
+		if(!runCheck())
+			die("Document service start failed. Click back to return home."); // nuts! give up
+	}
+}
+else{
+	// maybe next time
+	die("Cda generation service turned off: Verify in Administration->Globals! Click back to return home."); // Die an honorable death!!
+}
+
+//eventually below will qualify what document to fetch
+$parameterArray = array();
+$parameterArray ['encounter'];
+$parameterArray ['combination'] = $pid;
+$parameterArray ['components']; // = 'progress_note|consultation_note|continuity_care_document|diagnostic_image_reporting|discharge_summary|history_physical_note|operative_note|procedure_note|unstructured_document';
+$parameterArray ['sections']; // = 'allergies|medications|problems|immunizations|procedures|results|plan_of_care|vitals|social_history|encounters|functional_status|referral|instructions';
+$parameterArray ['downloadccda']=1;
+$parameterArray ['sent_by'];
+$parameterArray ['send'];
+$parameterArray ['view'] = 1;
+$parameterArray ['recipients'] = 'patient'; // emr_direct or hie else if not set $_SESSION['authUserID']
+$parameterArray [0] [6] = $_SESSION ['portal_username']; // set to an onsite portal user
+
+if(!isset($_SESSION ['site_id'])) $_SESSION ['site_id'] = 'default'; // do believe globals does this but I go rogue at times.
+$server_url = 'http://localhost'. $GLOBALS['webroot'];  // I alias into openemr directory on my sights causing webroot to be empty.
```

Comment:
Is there a way to figure out if the user on the portal is using http or https along with the server name, since would be good to use https if that is being used (especially if passing credentials) and would need to work from outside localhost also (unless I am missing something).